src/Tools/ExternalAccess/Razor/DocumentServiceProviderFactory.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.Razor
+{
+    internal abstract class DocumentServiceProviderFactory
```

Comment:
>DocumentServiceProviderFactory [](start = 28, length = 30)

Can we call this `RazorDocumentServiceProviderFactory`? We generally prefix the ExternalAccess types with the name of the partner component, so that it's clear it's not a Roslyn concept.